src/npyffi/mod.rs

```diff
@@ -34,16 +39,55 @@ fn get_numpy_api<'py>(
     Ok(api)
 }
 
+fn is_numpy_2<'py>(py: Python<'py>) -> bool {
```

Comment:
I'm just going to make this public, as I think users can benefit from it to avoid panics.

```suggestion
/// Returns whether the runtime `numpy` version is 2.0 or greater.
pub fn is_numpy_2<'py>(py: Python<'py>) -> bool {
```